3p/integration.js

```diff
@@ -375,21 +387,22 @@ const defaultAllowedTypesInCustomFrame = [
 
 
 /**
- * Gets data encoded in iframe name attribute.
- * @return {!JsonObject}
+ * Initialize 3p frame.
+ * @param {!Window} win
  */
-function getData(iframeName) {
-  try {
-    // TODO(bradfrizzell@): Change the data structure of the attributes
-    //    to make it less terrible.
-    return parseJson(iframeName)['attributes'];
-  } catch (err) {
-    if (!getMode().test) {
-      dev().info(
-          'INTEGRATION', 'Could not parse context from:', iframeName);
-    }
-    return FALLBACK_CONTEXT_DATA;
-  }
+function init(win) {
+  const data = getData();
+
+  win.AMP_3P_INIT = dict({
+    'mode': data['_context'].mode,
+  });
+
+  initLogConstructor();
+  setReportError(console.error.bind(console));
+
+  setExperimentToggles(data['_context'].experimentToggles);
+
+  delete win.AMP_3P_INIT;
```

Comment:
so this code has an implicit dependency that  initLogConstructor() needs to call getMode(), which is true for now, but it will be safer to explicitly call it out before you delete this data?

Actually, instead of storing the data as global, what do you think if we pass it along to `getMode(win, mode)`.